core/server/services/users.js

```diff
@@ -17,17 +19,66 @@ const path = require('path');
  * @prop {Object} ApiKey.NotFoundError
  * @prop {Object} User
  * @prop {(Object) => Promise} User.destroy
+ * @prop {(Object) => Promise} User.findAll
+ * @prop {Object} Session
+ * @prop {(Object) => Promise} Session.findAll
+ */
+
+/**
+ * @typedef {Object} IAuth
+ * @prop {Object} setup
+ * @prop {(isComplete: boolean) => () => Promise} setup.assertSetupCompleted
+ * @prop {Object} passwordreset
+ * @prop {(email: string, apiSettings: Object) => Promise<string>} passwordreset.generateToken
+ * @prop {(token: string, apiMail: Object) => Promise} passwordreset.sendResetNotification
  */
 
 class Users {
     /**
      * @param {Object} dependencies
      * @param {IdbBackup} dependencies.dbBackup
      * @param {IModels} dependencies.models
+     * @param {IAuth} dependencies.auth
+     * @param {Object} dependencies.apiMail
+     * @param {Object} dependencies.apiSettings
      */
-    constructor({dbBackup, models}) {
+    constructor({dbBackup, models, auth, apiMail, apiSettings}) {
         this.dbBackup = dbBackup;
         this.models = models;
+        this.auth = auth;
+        this.apiMail = apiMail;
+        this.apiSettings = apiSettings;
+    }
+
+    static async randomPassword() {
+        const bytes = await randomBytes(64);
+        return bytes.toString('hex');
+    }
+
+    async resetAllPasswords(frameOptions) {
+        return this.models.Base.transaction(async (t) => {
+            frameOptions.transacting = t;
+
+            // Reset all passwords
+            const users = await this.models.User.findAll({...frameOptions});
+            for (const user of users) {
+                await user.save({
+                    password: await Users.randomPassword(),
```

Comment:
Do we even need to set a random password or just setting the locked status is enough?

Reply:
good catch, the locked status should be enough!